### PR TITLE
Rework the getting-started sections of the docs.

### PR DESCRIPTION
**Pull request by mitechie**
- Make the JAAS the default getting-started
- Move the old one to a tut-lxd page instead
- Update the wording and docs on both pages to suit.

ToDo:

- Add a common "install juju client" docs page that all the rest can copy or
refer to.
- Bring the language and links to each of the pages into the same language.
- Run through the instructions step by step with juju 2.1 and 2.2 and verify
it's all working properly, update any screenshots.

### CODE REVIEW

**Comment by mitechie** [comment]
Added the start of an install doc but honestly I think that the reference-releases doc is really just an install doc like this. Maybe the best thing to do is to combine the two into a single reference-install.html and garden the two together.

File: src/en/tut-lxd.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
+Title: Getting started with Juju and LXD
+
+# Getting started with Juju and LXD
+
+[LXD](lxd-upstream) provides an amazing experience using machine containers to be able to
```
Comment (evilnick) [inline src/en/tut-lxd.md]: This first sentence seems to conflate a number of things and is difficult to grasp. I would suggest something like 

You may already know that [LXD] provides an amazing experience for managing lightweight containers. Juju can make use of LXD to provide a cloud-like experience from your local machine - etc etc


Possibly a short bullet list of why you would use it. We don't want to be too wordy in the beginning

File: src/en/tut-lxd.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+create the same models that are created with Juju on other public and private
+clouds. Using LXD as a method to test, verify, and replicate complex software
+deployments is a powerful tool that every Juju user needs. These instructions
+will get you up and running and deliver the best-possible experience with
+Juju. At the moment, that means using the latest release of Ubuntu with '[Long
```
Comment (evilnick) [inline src/en/tut-lxd.md]: Sometimes the markdown parser can choke on links with linebreaks in them. It is pretty much always better to begin a new line for long links, so:

      '[Long Term Support][long-term-support]' (LTS): 
      [16.04 LTS (Xenial)][Xenial-download]

File: src/en/tut-lxd.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+Term Support][long-term-support]' (LTS): [16.04 LTS
+                                          (Xenial)][Xenial-download].
+
+Our system will need the following:
```
Comment (evilnick) [inline src/en/tut-lxd.md]: change of POV. We usually use 'you/your' apart from exceptional cases (e.g. some example charms, which are 'ours')

So, 'Your system will also need the following:'

File: src/en/tut-lxd.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+Begin by installing the required software. You'll need Juju and LXD on the
+machine. This means that using LXD is not supported on the breadth of Juju
+supported operating systems.
+
+For more options in installing Juju see the [install docs][install].
+
+
+```bash
+sudo snap install --classic juju
```
Comment (evilnick) [inline src/en/tut-lxd.md]: we don't need to install Juju. This is a tutorial on setting up LXD for use with Juju. I think at some point we have to assume that people have actually installed Juju, either from 'getting started or the install page'. We do still need to install  lxd and zfsutils-linux though. 

Perhaps this could be solved simply by supplying a link to install the Juju client in the paragraph above?

File: src/en/tut-lxd.md [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+sudo apt install juju lxd zfsutils-linux
```
Comment (evilnick) [inline src/en/tut-lxd.md]: no longer needs Juju. Also, since we no longer have the ppa line, we should `sudo apt update` before running this

File: src/en/tut-lxd.md [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+a ZFS pool and appropriate networking for your LXD containers. To start this
+process, enter:
+
+```bash
+sudo lxd init
+```
+
+You will be asked several questions to configure LXD for use. Pressing Enter
+will accept the default answer (provided in square brackets). Only one answer,
+the size of the loop device in the below example, uses a non-default value.
```
Comment (evilnick) [inline src/en/tut-lxd.md]: I don't know if it is worth mentioning, but the default value for the loop device size is now dynamic (it is based on a % of storage available). I think it probably isn't worth it, but what do you think?

File: src/en/tut-lxd.md (hunk 4 above)
Comment (mitechie) [inline src/en/tut-lxd.md]: Since lxd is only going to be ubuntu based atm and we do have the install docs link above I figured that it was safe enough to include here for the sake of completeness.

File: src/en/tut-lxd.md (hunk 5 above)
Comment (mitechie) [inline src/en/tut-lxd.md]: Why do we need to update before this?

File: src/en/tut-lxd.md (hunk 6 above)
Comment (mitechie) [inline src/en/tut-lxd.md]: wfm, removing.

File: src/en/tut-lxd.md (hunk 1 above)
Comment (mitechie) [inline src/en/tut-lxd.md]: Reworded, thanks for suggestion. I like the phrase "cloud-like experience" in particular.

File: src/en/tut-lxd.md (hunk 2 above)
Comment (mitechie) [inline src/en/tut-lxd.md]: updated

**Comment by mitechie** [comment]
Thanks for the review. I've updated based on comments and asked a couple of questions in the review points.

File: src/en/tut-lxd.md [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+Congratulations, you have just deployed an application with Juju!
+
+!!! Note: To remove all the applications in the model you just created, it is
+often quickest to destroy the model with the command
+`juju destroy-model default` and then [create a new model][models].
+
+
+## Next Steps
+
+Now that you have a Juju-powered cloud, it is time to explore the amazing
```
Comment (evilnick) [inline src/en/tut-lxd.md]: I think this should also be updated to reflect that this is no longer 'getting-started' . Perhaps something like

**Now that you have configured Juju to work with LXD images, you can test, tweak, develop and experiment with Juju charms rapidly, easily and without additional costs.** 

or similar

File: src/en/reference-install.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+
+
+
+## Getting development releases
+
+The Juju team regularly puts out alpha, beta, and RC releases during a
+development cycle. We encourage users to please test these versions out with
+their workloads and use cases and [file bugs][bugs] if you hit any issues.
+Remember that while in development feedback on usability and missing
```
Comment (evilnick) [inline src/en/reference-install.md]: comma after 'development'

File: src/en/reference-install.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+functionality is very important so don't just file bugs for things that go
+wrong, but also for things that have an opportunity to be "more right".
+
+
+## Ubuntu
+
+It is possible to install other versions, including beta releases of
+Juju via a snap package. See the [install page][install] for more information.
```
Comment (evilnick) [inline src/en/reference-install.md]: isn't this a circular reference?

File: src/en/reference-install.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+```
+
+Or you can get a daily build using the edge channel.
+
+```bash
+   snap install <snap> --edge
+```
+
+
+## macOS
```
Comment (evilnick) [inline src/en/reference-install.md]: should be ###

File: src/en/reference-install.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+
+The Juju team regularly puts out alpha, beta, and RC releases during a
+development cycle. We encourage users to please test these versions out with
+their workloads and use cases and [file bugs][bugs] if you hit any issues.
+Remember that while in development feedback on usability and missing
+functionality is very important so don't just file bugs for things that go
+wrong, but also for things that have an opportunity to be "more right".
+
+
+## Ubuntu
```
Comment (evilnick) [inline src/en/reference-install.md]: should be ###

File: src/en/reference-install.md [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+   brew install --devel juju
+```
+
+Or you can use the binary built for macOS:
+: [juju-core_2.2-beta3-osx.tar.gz](https://launchpad.net/juju/2.2/2.2-beta3/+download/juju-2.2-beta3-osx.tar.gz) ([md5](https://launchpad.net/juju/2.2/2.2-beta3/+download/juju-2.2-beta3-osx.tar.gz/+md5))
+{: .instruction }
+
+
+
+## Windows
```
Comment (evilnick) [inline src/en/reference-install.md]: should be ###

File: src/en/reference-install.md [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+
+The latest windows .exe can be downloaded from the Juju download page [https://launchpad.net/juju/+download](https://launchpad.net/juju/+download)
+
+The latest is version 2.2-beta3 located at:
+: [juju-setup-2.2-beta3.exe](https://launchpad.net/juju/2.2/2.2-beta3/+download/juju-setup-2.2-beta3.exe) ([md5](https://launchpad.net/juju/2.2/2.2-beta3/+download/juju-setup-2.2-beta3.exe/+md5))
+{: .instruction }
+
+
+
+# Building from source
```
Comment (evilnick) [inline src/en/reference-install.md]: should be ##

File: src/en/reference-install.md [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+
+A Windows installer is available for Juju.
+
+[juju-setup-2.1.2.exe](https://launchpad.net/juju/2.1/2.1.2/+download/juju-setup-2.1.2.exe)([md5](https://launchpad.net/juju/2.1/2.1.2/+download/juju-setup-2.1.2.exe/+md5))
+
+
+### Centos and other Linuxes
+
+
+Spans enable installing the the Juju client on an array of Linux
```
Comment (evilnick) [inline src/en/reference-install.md]: Snaps

File: src/en/reference-install.md (hunk 9 above)
Comment (mitechie) [inline src/en/reference-install.md]: Heh, I think I copied it from the other page and missed cleaning it up.

File: src/en/tut-lxd.md (hunk 4 above)
Comment (mitechie) [inline src/en/tut-lxd.md]: I've updated the text around there.

File: src/en/tut-lxd.md (hunk 7 above)
Comment (mitechie) [inline src/en/tut-lxd.md]: Updated with new text.

File: src/en/reference-install.md (hunk 14 above)
Comment (mitechie) [inline src/en/reference-install.md]: ty

File: src/en/reference-install.md (hunk 8 above)
Comment (mitechie) [inline src/en/reference-install.md]: ty

File: src/en/reference-install.md (hunk 11 above)
Comment (mitechie) [inline src/en/reference-install.md]: ty

File: src/en/reference-install.md (hunk 10 above)
Comment (mitechie) [inline src/en/reference-install.md]: ty

File: src/en/reference-install.md (hunk 12 above)
Comment (mitechie) [inline src/en/reference-install.md]: ty

File: src/en/reference-install.md (hunk 13 above)
Comment (mitechie) [inline src/en/reference-install.md]: ty

**Comment by mitechie** [comment]
Thanks, I've updated per the latest review comments.

**Comment by evilnick** [comment]
Thanks for the speedy updates. I'd also like @degville to have a look over this in case there are bits I missed

File: src/en/getting-started.md [hunk 15]
```diff
@@ -1,255 +1,323 @@
...
 
-These instructions will get you up and running and deliver the best-possible
-experience with Juju. At the moment, that means using the latest release of
-Ubuntu with '[Long Term Support][long-term-support]' (LTS): [16.04 LTS (Xenial)][Xenial-download].
+Juju works across public clouds, private clouds, and locally with LXD based
+deployment and operating of applications. Juju works through a controller that
+will process all of the events that occur. These instructions will walk you
+through using the hosted Juju as a Service (JAAS) controller and requires you
+to have public cloud credentials ready. If you'd prefer to test it out
+locally, we've got a tutorial on setting up a [LXD based controller over
```
Comment (degville) [inline src/en/getting-started.md]: 'over here' sounds a little too casual to me, although I appreciate you're trying to be less formal. We'd normally write something more like `see our _Getting started with Juju and LXD_ guide further details`.

File: src/en/getting-started.md [hunk 16]
```diff
@@ -1,255 +1,323 @@
...
+You will see a list of available charms and bundles with a description of
+each. Select a [charm][charms] or [bundle][bundles] to learn more about it.
 
-This will return a list of the controllers known to Juju, which at the moment is
-the one we just created:
-  
-```no-highlight
-Use --refresh flag with this command to see the latest information.
+When you have selected a charm or bundle (hint: to see how easy JAAS makes
+complex deployments, try the Canonical Kubernetes bundle!) it can be added
```
Comment (degville) [inline src/en/getting-started.md]: I think we should add a link to https://jujucharms.com/canonical-kubernetes/

File: src/en/getting-started.md [hunk 17]
```diff
@@ -1,255 +1,323 @@
...
+to your model by pressing the 'Add to model' button...
```
Comment (degville) [inline src/en/getting-started.md]: It may be more consistent to replace the ellipsis with a colon.

File: src/en/getting-started.md [hunk 18]
```diff
@@ -1,255 +1,323 @@
...
+inspector on the left will disappear, to show that everything is working as
+expected.
+
+!!! Tip: You can check and manage existing models through the JAAS GUI by
+clicking on your username
+
+## Use the command line
+
+Juju can also be used from the command line. Models you've created in JAAS are
+also able to be operated against via the CLI and vice versa. In order to use
```
Comment (degville) [inline src/en/getting-started.md]: this sounds clunky to me. How about, `Models created with JAAS can also be operated on from the CLI, and vice versa - models created on the CLI can be modified by the GUI.`

File: src/en/getting-started.md [hunk 19]
```diff
@@ -1,255 +1,323 @@
...
+the command line, you will first need to install the Juju client software on
+your machine.
+
+Juju is available for various types of Linux, macOS, and Windows.
+Click on the sections below for the relevant instructions or visit the
+[install instructions][install] for a thorough run down of your options.
+
+To most quickly get started pick your OS from below.
```
Comment (degville) [inline src/en/getting-started.md]: I'd suggest `Get started quickly by picking your OS from below:`

File: src/en/getting-started.md [hunk 20]
```diff
@@ -1,255 +1,323 @@
...
+
+^# On an OS which supports snap packages
+
+   For Ubuntu 16.04 LTS (Xenial) and other OSes which have snaps enabled
+   ([see the snapcraft.io site][snapcraft]) you can install the latest
+   stable release of Juju with the command:
+
+         sudo snap install juju --classic
+
+   You can check which version of Juju you have installed with
```
Comment (degville) [inline src/en/getting-started.md]: with: (colon)

File: src/en/getting-started.md [hunk 21]
```diff
@@ -1,255 +1,323 @@
...
+
+   A pre-compiled binary is available for CentOS.
+
+   Please see the [install docs][install] for a link to the latest
+   version.
+
+
+## Register or login to JAAS
+
+To connect to JAAS from the command line you'll need to login. Enter the following command:
```
Comment (degville) [inline src/en/getting-started.md]: on the command line, _login_ could be ambiguous. How about `authenticate the CLI against JAAS with the following command:`

File: src/en/getting-started.md [hunk 22]
```diff
@@ -1,255 +1,323 @@
...
+This command will open a new window in your default web browser and use
+[Ubuntu SSO][ubuntusso] to authorise your account. If the browser doesn't open,
+you can manually copy and paste the unique authorisation URL from the command
+output.
 
-![juju status](./media/juju-mediawiki-status.png)
+After successful authentication, you will be asked to enter a descriptive name
+for the JAAS controller (we suggest JAAS). Now you can operate your models or
+applications in JAAS from the command line. You can see a list of all of your
+controllers using the cli command:
```
Comment (degville) [inline src/en/getting-started.md]: Uppercase CLI was used earlier (we normally write `command line`).

File: src/en/getting-started.md [hunk 23]
```diff
@@ -1,255 +1,323 @@
...
+This process may take a few minutes to complete, as Juju releases
+the running resources back to the cloud.
+
+## Next steps
+
+Congratulations! You have now deployed a complex workload in the cloud without
+hours of looking up config options or wrestling with install scripts!
+
+To discover more about what Juju can do for you, we suggest some of the
+following pages of documentation
```
Comment (degville) [inline src/en/getting-started.md]: documentation: (colon)

File: src/en/jaas-cli.md [hunk 24]
```diff
@@ -21,24 +21,24 @@ details  {
...
-   For Ubuntu 16.04 LTS (Xenial) and other OSes which have snaps enabled 
+   For Ubuntu 16.04 LTS (Xenial) and other OSes which have snaps enabled
    ([see the snapcraft.io site][snapcraft]) you can install the latest
    stable release of Juju with the command:
-   
+
          sudo snap install juju --classic
 
-   You can check which version of Juju you have installed with 
+   You can check which version of Juju you have installed with
```
Comment (degville) [inline src/en/jaas-cli.md]: with: (colon)

File: src/en/getting-started.md (hunk 15 above)
Comment (mitechie) [inline src/en/getting-started.md]: reworded

File: src/en/getting-started.md (hunk 16 above)
Comment (mitechie) [inline src/en/getting-started.md]: +1 added

File: src/en/getting-started.md (hunk 17 above)
Comment (mitechie) [inline src/en/getting-started.md]: reworded

File: src/en/getting-started.md (hunk 18 above)
Comment (mitechie) [inline src/en/getting-started.md]: reworded

File: src/en/getting-started.md (hunk 19 above)
Comment (mitechie) [inline src/en/getting-started.md]: ty, updated

File: src/en/getting-started.md (hunk 20 above)
Comment (mitechie) [inline src/en/getting-started.md]: ty

File: src/en/getting-started.md (hunk 21 above)
Comment (mitechie) [inline src/en/getting-started.md]: reworded

File: src/en/getting-started.md (hunk 22 above)
Comment (mitechie) [inline src/en/getting-started.md]: updated cli to command line throughout

File: src/en/getting-started.md (hunk 23 above)
Comment (mitechie) [inline src/en/getting-started.md]: ty

File: src/en/jaas-cli.md (hunk 24 above)
Comment (mitechie) [inline src/en/jaas-cli.md]: ty

**Comment by mitechie** [comment]
Pushed an update addressing the changes requested by @degville

File: src/en/getting-started.md [hunk 25]
```diff
@@ -157,16 +158,16 @@ clicking on your username
 
 ## Use the command line
 
-Juju can also be used from the command line. Models you've created in JAAS are
-also able to be operated against via the CLI and vice versa. In order to use
-the command line, you will first need to install the Juju client software on
-your machine.
+Juju can also be used from the command line. Models you've created with the
+JAAS website can also able to be operated via the normal Juju command line. In
```
Comment (evilnick) [inline src/en/getting-started.md]: I think if you delete 'able to' here, we're golden

File: src/en/getting-started.md [hunk 26]
```diff
@@ -251,7 +253,8 @@ output.
 After successful authentication, you will be asked to enter a descriptive name
 for the JAAS controller (we suggest JAAS). Now you can operate your models or
 applications in JAAS from the command line. You can see a list of all of your
-controllers using the cli command:
+controllers using the command line command:
```
Comment (evilnick) [inline src/en/getting-started.md]: I would just put 'command:' or 'command line:'

Comment (evilnick) [inline src/en/getting-started.md]: actually, I prefer just 'command:'

Comment (mitechie) [inline src/en/getting-started.md]: ty for the catch.

File: src/en/getting-started.md (hunk 25 above)
Comment (mitechie) [inline src/en/getting-started.md]: agree, remoed

**Comment by mitechie** [comment]
Updated, ty for the reviews. Please let me know if there's anything else that catches your eyes.

**Comment by evilnick** [comment]
This looks okay now, but we will have to sort out the navigation due to pages removed/moved. Do you want to do that here?

**Comment by mitechie** [comment]
Ah, that's a good point. I'd not looked at the nav setup. Might as well do it here to test it out and make sure it all works with the changes/file moves here. 

I'll update shortly.

**Comment by evilnick** [comment]
Cool. You'll find the 'navigation.tpl' in the src folder. It's just html so reasonably straightforward

**Comment by mitechie** [comment]
Is the nav.txt in the tools still used?

**Comment by mitechie** [comment]
I've updated the nav, corrected merge conflicts with the branch from upstream, and fixed a couple of links.

**Comment by mitechie** [comment]
@evilnick I'm hitting https://www.bountysource.com/issues/36261101-python-3-syntaxerror-from-long-literal-0x950412del-in-po-msgfmt-py trying to get the linkchecker installed and test out the links. Can you run it and let me know what needs to be corrected? I'm assume I've got to have missed something in the changes.

File: src/navigation.tpl [hunk 27]
```diff
@@ -1,16 +1,10 @@
...
+    <li class="section"><h4 class="header toggle-target">Quick Guides</h4>
         <ul>
-        <li class="section"><a class="header" href="getting-started.html">First use!</a>
-            <ul class="sub">
-                <li><a href="getting-started-general.html">Install for non-16.04</a></li>
-            </ul>
-        </li>
-        <li><a href="getting-started-jaas.html">Juju as a Service (JAAS)</a></li>
+        <li class="section"><a class="header" href="getting-started.html">First use!</a></li>
+        <li><a href="tut-lxd.html">Local Juju with LXD cloud</a></li>
```
Comment (evilnick) [inline src/navigation.tpl]: I think we can come up with a better name for this. For one thing, LXD isn't the benefit, so it is rather irrelevant to messaging (LXD is how we deliver what they want).
Alternative suggestions - Juju on a no-cost, local cloud

**Comment by evilnick** [comment]
@mitechie yeah, the linkchecker is a pain - such a useful tool, but pretty much abandoned now. It doesn't work with Python3 at all. Are you trying to get it working on a Mac? It works on xenial with 2.7 and 3 installed

**Comment by evilnick** [comment]
@mitechie yup, the following help pages still reference getting-started-jaas:

 help-azure
 help-google
 help-aws

File: src/navigation.tpl [hunk 28]
```diff
@@ -1,16 +1,10 @@
  <ul>
     <li><a href="about-juju.html">About Juju</a></li>
-    <li><a href="introducing-2.html">Introducing Juju 2.0!</a></li>
-    <li><a href="whats-new.html">What's new in 2.1</a></li>
 <!-- SECTION -->
-    <li class="section"><h4 class="header toggle-target">Get Started</h4>
+    <li class="section"><h4 class="header toggle-target">Quick Guides</h4>
```
Comment (mitechie) [inline src/navigation.tpl]: I guess I've been in too much "getting started" all over the place and as we're starting to refer to things as tutorials or guides I thought it might be a good update to reduce using "getting" so much any more. I'm +1 on moving the First use to the new "Get started". I almost did that in here.

File: src/navigation.tpl (hunk 27 above)
Comment (mitechie) [inline src/navigation.tpl]: I think that it's worth keeping the lxd branding going. I worry about it being too generic and folks heading here that don't have or can't use LXD on their systems.

File: src/navigation.tpl [hunk 29]
```diff
@@ -195,14 +189,16 @@
 <!-- SECTION -->
     <li class="section"><h4 class="header toggle-target">Reference</h4>
         <ul>
+        <li><a href="reference-install.html">Install Juju</a></li>
+        <li><a href="introducing-2.html">Introducing Juju 2.0!</a></li>
+        <li><a href="whats-new.html">What's new in 2.1</a></li>
```
Comment (mitechie) [inline src/navigation.tpl]: I'm +1 when we do the what's new with 2.2 at the top? I guess I feel like there's a time window here and 2.1 is old enough that it pushes more important stuff down.

**Comment by mitechie** [comment]
Updated the name of the nav link for Get started and updated the broken link pages notes. Thanks for checking those.

File: src/navigation.tpl (hunk 28 above)
Comment (evilnick) [inline src/navigation.tpl]: Yes, it is a pain, but it is also a common signpost, so I think it is useful to keep somewhere!

File: src/navigation.tpl (hunk 27 above)
Comment (evilnick) [inline src/navigation.tpl]: hmmm. Yes, that is an issue. I don't think its worth holding up the rest of the updates for though, but I think we should work on a better name

**Comment by evilnick** [comment]
Ok, I'll merge this now. We will have to backport all the changes to get it into 2.1, 
I will take a stab at that tomorrow

**Comment by mitechie** [comment]
Thank you for working with me on the updates. Appreciate all the sanity checking and time spent.

**Comment by evilnick** [comment]
NP, thanks for all the work you did!
